src/components/OrderBookWidget.tsx

```diff
@@ -224,103 +223,257 @@ function _printOrderBook(pricePoints: PricePointDetails[], baseToken: TokenDetai
   })
 }
 
-const draw = (
-  chartElement: HTMLElement,
-  baseToken: TokenDetails,
-  quoteToken: TokenDetails,
-  networkId: number,
-  hops?: number,
-): am4charts.XYChart => {
-  const baseTokenLabel = safeTokenName(baseToken)
-  const quoteTokenLabel = safeTokenName(quoteToken)
-  const market = baseTokenLabel + '-' + quoteTokenLabel
-  am4core.useTheme(am4themesSpiritedaway)
-  am4core.options.autoSetClassName = true
-  const chart = am4core.create(chartElement, am4charts.XYChart)
-  const networkDescription = networkId !== Network.Mainnet ? `${getNetworkFromId(networkId)} ` : ''
-
-  // Add data
-  chart.dataSource.url = dexPriceEstimatorApi.getOrderBookUrl({
-    baseTokenId: baseToken.id,
-    quoteTokenId: quoteToken.id,
-    hops,
-    networkId,
-  })
-  chart.dataSource.adapter.add('parsedData', data => {
-    try {
-      const bids = processData(data.bids, baseToken, quoteToken, Offer.Bid)
-      const asks = processData(data.asks, baseToken, quoteToken, Offer.Ask)
-      const pricePoints = bids.concat(asks)
+// --- Initial zoom calculation helper functions ---
+function calcRange(minBid: number, maxBid: number, minAsk: number, maxAsk: number): number {
+  return Math.max(maxBid, maxAsk) - Math.min(minBid, minAsk)
+}
 
-      // Sort points by price
-      pricePoints.sort((lhs, rhs) => lhs.price.comparedTo(rhs.price))
+function calcSpread(maxBid: number, minAsk: number): number {
+  return Math.abs(minAsk - maxBid)
+}
 
-      _printOrderBook(pricePoints, baseToken, quoteToken)
+function calcZoomInterval(spread: number): number {
+  // TODO: magic number, move to const
+  return spread * 2
+}
 
-      return pricePoints
-    } catch (error) {
-      console.error('Error processing data', error)
-      return []
-    }
-  })
+function calcLowerZoomX(minBid: number, maxBid: number, minAsk: number, zoomInterval: number): number {
+  return Math.max(Math.min(maxBid, minAsk) - zoomInterval, minBid)
+}
+
+function calcUpperZoomX(minAsk: number, maxAsk: number, maxBid: number, zoomInterval: number): number {
+  return Math.min(Math.max(maxBid, minAsk) + zoomInterval, maxAsk)
+}
 
-  // Colors
-  const colors = {
-    green: '#3d7542',
-    red: '#dc1235',
+function calcUpperZoomY(
+  bids: PricePointDetails[],
+  asks: PricePointDetails[],
+  lowerZoomX: number,
+  upperZoomX: number,
+): number {
+  let bidsVolume = 0
+  for (let i = 0; i < bids.length; i++) {
+    if (lowerZoomX < bids[i].priceNumber) {
+      bidsVolume = bids[i].totalVolumeNumber
+      break
+    }
+  }
+  let asksVolume = asks.length > 0 ? asks[0].totalVolumeNumber : 0
+  for (let i = 0; i < asks.length; i++) {
+    if (upperZoomX < asks[i].priceNumber) {
+      break
+    }
+    asksVolume = asks[i].totalVolumeNumber
   }
+  console.log(`bidsVolume: ${bidsVolume}; asksVolume: ${asksVolume}`)
+  return Math.max(bidsVolume, asksVolume)
+}
+
+function calcMinX(minBid: number, minAsk: number): number {
```

Comment:
it seems a bit strange to create a wrapper of MAX and MIN functions

Reply:
To keep everything tidy, one function to calculate every parameter.